opal/mca/btl/iwarp/btl_iwarp.h

```diff
@@ -0,0 +1,927 @@
+    mca_base_var_source_t receive_queues_source;
+
+    /** Colon-delimited list of filenames for device parameters */
+    char *device_params_file_names;
+
+    /** Whether we're in verbose mode or not */
+    bool verbose;
+
+    /** Whether we want a warning if no device-specific parameters are
+        found in INI files */
+    bool warn_no_device_params_found;
+    /** Whether we want a warning if non default GID prefix is not configured
+        on multiport setup */
+    bool warn_default_gid_prefix;
+    /** Whether we want a warning if the user specifies a non-existent
+        device and/or port via btl_iwarp_if_[in|ex]clude MCA params */
+    bool warn_nonexistent_if;
+    /** Whether we want to abort if there's not enough registered
+        memory available */
+    bool abort_not_enough_reg_mem;
+
+    /** Dummy argv-style list; a copy of names from the
+        if_[in|ex]clude list that we use for error checking (to ensure
+        that they all exist) */
+    char **if_list;
+    bool use_message_coalescing;
+    unsigned int cq_poll_ratio;
+    unsigned int cq_poll_progress;
+    unsigned int cq_poll_batch;
+    unsigned int eager_rdma_poll_ratio;
+    int rdma_qp;
+    int credits_qp; /* qp used for software flow control */
+    bool cpc_explicitly_defined;
+    /**< free list of frags only; used for pining user memory */
+    opal_free_list_t send_user_free;
+    /**< free list of frags only; used for pining user memory */
+    opal_free_list_t recv_user_free;
+    /**< frags for coalesced massages */
+    opal_free_list_t send_free_coalesced;
+    /** Default receive queues */
+    char* default_recv_qps;
+    /** GID index to use */
+    int gid_index;
+    /*  Whether we want to allow connecting processes from different subnets.
+     *  set to 'no' by default */
+    bool allow_different_subnets;
+    /** Whether we want a dynamically resizing srq, enabled by default */
+    bool enable_srq_resize;
+    bool allow_max_memory_registration;
+    int memory_registration_verbose_level;
+    int memory_registration_verbose;
+    int ignore_locality;
+#if OPAL_CUDA_SUPPORT
+    bool cuda_async_send;
+    bool cuda_async_recv;
+    bool cuda_have_gdr;
+    bool driver_have_gdr;
+    bool cuda_want_gdr;
+#endif /* OPAL_CUDA_SUPPORT */
+#if HAVE_DECL_IBV_LINK_LAYER_ETHERNET
+    bool rroce_enable;
+#endif
+    unsigned int num_default_gid_btls; /* numbers of btl in the default subnet */
+}; typedef struct mca_btl_iwarp_component_t mca_btl_iwarp_component_t;
+
+OPAL_MODULE_DECLSPEC extern mca_btl_iwarp_component_t mca_btl_iwarp_component;
+
+typedef mca_btl_base_recv_reg_t mca_btl_iwarp_recv_reg_t;
+
+/**
+ * Common information for all ports that is sent in the modex message
+ */
+typedef struct mca_btl_iwarp_modex_message_t {
+    /** The subnet ID of this port */
+    uint64_t subnet_id;
+    /** LID of this port */
+    uint16_t lid;
+    /** APM LID for this port */
+    uint16_t apm_lid;
```

Comment:
I think you can remove some stuff from the modex message, too (e.g., APM -- possibly others).